crates/noirc_frontend/src/hir/type_check/stmt.rs

```diff
@@ -181,13 +181,13 @@ fn type_check_lvalue(
             let index_type = type_check_expression(interner, &index, errors);
             let expr_span = interner.expr_span(&index);
 
-            index_type.unify(&Type::comp_time(Some(expr_span)), expr_span, errors, || {
-                TypeCheckError::TypeMismatch {
-                    expected_typ: "comptime Field".to_owned(),
+            if index_type.is_subtype_of(&Type::field(Some(expr_span)), expr_span).is_err() {
```

Comment:
Same here with regards to make_subtype_of

Reply:
Done